kubeconfig/kubeconfig.go

```diff
@@ -0,0 +1,127 @@
+//
+// Copyright (c) 2019-2020 Red Hat, Inc.
+// This program and the accompanying materials are made
+// available under the terms of the Eclipse Public License 2.0
+// which is available at https://www.eclipse.org/legal/epl-2.0/
+//
+// SPDX-License-Identifier: EPL-2.0
+//
+// Contributors:
+//   Red Hat, Inc. - initial API and implementation
+//
+package kubeconfig
+
+import (
+	"io/ioutil"
+	"net"
+	"os"
+
+	"github.com/eclipse/che-machine-exec/exec"
+	"github.com/sirupsen/logrus"
+	"gopkg.in/yaml.v2"
+)
+
+type KubeConfig struct {
+	APIVersion     string     `yaml:"apiVersion"`
+	Clusters       []Clusters `yaml:"clusters"`
+	Users          []Users    `yaml:"users"`
+	Contexts       []Contexts `yaml:"contexts"`
+	CurrentContext string     `yaml:"current-context"`
+	Kind           string     `yaml:"kind"`
+}
+
+type Clusters struct {
+	Cluster ClusterInfo `yaml:"cluster"`
+	Name    string      `yaml:"name"`
+}
+
+type ClusterInfo struct {
+	CertificateAuthority string `yaml:"certificate-authority"`
+	Server               string `yaml:"server"`
+}
+
+type Users struct {
+	Name string `yaml:"name"`
+	User User   `yaml:"user"`
+}
+
+type User struct {
+	Token string `yaml:"token"`
+}
+
+type Contexts struct {
+	Context Context `yaml:"context"`
+	Name    string  `yaml:"name"`
+}
+
+type Context struct {
+	Cluster   string `yaml:"cluster"`
+	Namespace string `yaml:"namespace"`
+	User      string `yaml:"user"`
+}
+
+func generateKubeConfig(token, server, namespace string) *KubeConfig {
+	return &KubeConfig{
+		APIVersion: "v1",
+		Clusters: []Clusters{
+			Clusters{
```

Comment:
Including type here is unnecessary (you can do `[]clusters{ {}, {} }`), and similar for `Users`, `Contexts`.